### DISCUSSION TITLE
URQL not rerending on completion

### DISCUSSION THREAD

**Discussion by Dhallagan** [urql-graphql/urql#2144] [Q&A] [answered]
New to URQL, working in react native and send a urql request. The native screen says loading but on completion the data is returned but the screen does not update the components.  I thought URQL was basically a useEffect?

What am I missing?  I expected to see a list of classes.

```

export default function ClassesScreen() {
  const navigation = useNavigation();
  const inset = useSafeAreaInsets();
  const [classes, setClasses] = useState([]);

  const [result] = useQuery({
    query: ClassesQuery,
  });

  let { data, fetching, error } = result;

  if (fetching) {
    return <Text style={{ color: "white" }}>Loading...</Text>;
  }
  if (error) {
    return <Text style={{ color: "white" }}>Error: {error.message}</Text>;
  }

  if (data) {
    return (
      <>
        <ClassHeader title={"Classes"} style={{ color: "white" }}>
          {/* <OutlinedText>Classes</OutlinedText> */}
        </ClassHeader>
        <View style={styles.tagsContainer}>
          <Chip title="  All  " outlined />
          <Chip title="Brand New" />
          <Chip title="Trending" />
          <Chip title="Popular" />
        </View>
        <ScrollView style={styles.container}>
          {data.classes.edges?.map((x) => (
            <ClassCard
              id={x.node.id}
              title={x.node.title}
              image={x.node.cover?.url}
            >
            </ClassCard>
          ))}
        </ScrollView>
      </>
    );
  }

```

**Comment by JoviDeCroock** [comment, marked as answer]
I mean, it's a lot more than just a "useEffect" 😅 but yeah it's hard to tell what you're doing wrong from this example as it's looking okay, you might want to `console.log` result to see whether there's something wrong i.e. you're getting an error, ...

**Reply by Dhallagan** [reply]
It works fine on react native web. And I can see the console return the data. It seems to be platform specific. Thank you though, just making sure it wasn't something obvious. Definitely a rerender issue iOS

**Reply by JoviDeCroock** [reply]
Haven't personally seen or heard of any issues in React-Native iOS so I don't know personally 😅 might relate to the ScrollView
